Machine Learning clustering analyses requests

koenderks opened the pull request:
As requested in https://github.com/jasp-stats/jasp-issues/issues/563

+ Add cluster means table
+ Add cluster density plots
+ Add cluster mean plots
+ Add percentage of explained within-cluster heterogeneity to cluster information table

koenderks committed:
Add cluster means table to machine learning

koenderks committed:
Add cluster density plots to machine learning

koenderks committed:
Add cluster means plot to machine learning

koenderks committed:
Add percentage of explained within cluster heterogeneity to cluster information table

koenderks committed:
Add line for Explained proportion within-cluster heterogeneity

+ update unit tests

koenderks committed:
Update jasp files and help files to include the latest changes

koenderks committed:
Add two additional options for the cluster means plot

Kucharssim commented on Resources/Help/analyses/mlclusteringdensitybased.md:
```diff
@@ -12,14 +12,17 @@ Density-based clustering is a soft clustering method where clusters are construc
 
 #### Plots
 - K-distance plot: Generates a plot with the nearest neighbors distance (the amount of nearest neighbors is determined by the Min. core points parameter) on the y-axis and the points sorted by distance on the x-axis. This plot can be used for determining the optimal Epsilon value. The value where the graph shows a kink represents the optimal Epsilon value (the y-axis value).
+- Cluster means: For each predictor variable, generates a plot showing the mean of each cluster and it's 95% confidence interval.
```
```suggestion
- Cluster means: For each predictor variable, generates a plot showing the mean of each cluster and its 95% confidence interval.
```

Kucharssim commented on Resources/Help/analyses/mlclusteringfuzzycmeans.md:
```diff
@@ -21,6 +22,8 @@ Fuzzy c-means clustering is a soft partitioning method that provides an output t
 
 #### Plots
 - Elbow method: Generates a plot with the total within sum of squares on the y-axis and the number of clusters on the x-axis. This plot can be used for determining the optimal number of clusters. The plot shows three curves using AIC, BIC, and 'elbow method' optimization.
+- Cluster means: For each predictor variable, generates a plot showing the mean of each cluster and it's 95% confidence interval.
```
```suggestion
- Cluster means: For each predictor variable, generates a plot showing the mean of each cluster and its 95% confidence interval.
```

Kucharssim commented on Resources/Help/analyses/mlclusteringhierarchical.md:
```diff
@@ -22,6 +23,8 @@ Hierarchical clustering is a hard partitioning algorithm which aims to partition
 #### Plots
 - Elbow method: Generates a plot with the total within sum of squares on the y-axis and the number of clusters on the x-axis. This plot can be used for determining the optimal number of clusters. The plot shows three curves using AIC, BIC, and 'elbow method' optimization.
 - Dendrogram: Generates a dendrogram of the clustering output. A dendrogram can be interpreted as a tree upside down. At the bottom, there are the leaves, and as one goes up the tree, leaves merge into branches (i.e., groups of observations). The leaves correspond to an observation. Also, the height at which leaves merge together indicates the (dis)similarity between observations or groups of obs ...
+- Cluster means: For each predictor variable, generates a plot showing the mean of each cluster and it's 95% confidence interval.
```
```suggestion
- Cluster means: For each predictor variable, generates a plot showing the mean of each cluster and its 95% confidence interval.
```

Kucharssim commented on Resources/Help/analyses/mlclusteringkmeans.md:
```diff
@@ -22,6 +23,8 @@ K-means clustering is a hard partitioning algorithm which aims to partition data
 
 #### Plots
 - Elbow method: Generates a plot with the total within sum of squares on the y-axis and the number of clusters on the x-axis. This plot can be used for determining the optimal number of clusters. The plot shows three curves using AIC, BIC, and 'elbow method' optimization.
+- Cluster means: For each predictor variable, generates a plot showing the mean of each cluster and it's 95% confidence interval.
```
```suggestion
- Cluster means: For each predictor variable, generates a plot showing the mean of each cluster and its 95% confidence interval.
```

Kucharssim commented on Resources/Help/analyses/mlclusteringrandomforest.md:
```diff
@@ -22,6 +23,8 @@ Random Forest clustering is a hard partitioning algorithm which aims to partitio
 
 #### Plots
 - Elbow method: Generates a plot with the total within sum of squares on the y-axis and the number of clusters on the x-axis. This plot can be used for determining the optimal number of clusters. The plot shows three curves using AIC, BIC, and 'elbow method' optimization.
+- Cluster means: For each predictor variable, generates a plot showing the mean of each cluster and it's 95% confidence interval.
```
```suggestion
- Cluster means: For each predictor variable, generates a plot showing the mean of each cluster and its 95% confidence interval.
```

Kucharssim reviewed (changes requested):
The plots look very nice! I did not find anything of importance in the code, I left some comments for the help files.

Not sure whether it’s relevant right now, but would it be possible to do the gettexting here, or will be there a different PR for this? 

Also, would it be possible to add unit tests for the new plots?

Kucharssim commented:
Oh yeah, and I would also suggest adding an option to select the % for the confidence intervals...

koenderks committed:
change tests

koenderks commented:
Thanks Simon! I have changed the help files and implemented unit tests for the plot. The confidence interval adjustments might be something for later.

koenderks committed:
Updated the unit tests
